build

```diff
@@ -191,11 +216,23 @@ for my $package (@packages) {
         next;
     }
 
-    if ($os eq 'mac' && !$manifest->{supports_macos}) {
-        print STDERR "skipping $package because it doesn't declare macOS support\n";
-        next;
+    if (exists $manifest->{supported_platforms}) {
+        @platforms_enabled = ();
+        for my $p (@{$manifest->{supported_platforms}}) {
+            push(@platforms_enabled, $p);
+        }
     }
 
+    my $package_enabled = 0;
+    foreach(@platforms_enabled) {
+        if ($_ eq "$os") {
+            $package_enabled = 1;
+        }
+    }
+    if ($package_enabled eq 0) {
+        print STDERR "skipping $package because it doesn't declare platform support\n";
```

Comment:
Would be less confusing if you write `Skipping the $package since it doesn't support current platform`